ui/job-view/pushes/PushList.jsx

```diff
@@ -34,14 +38,16 @@ export default class PushList extends React.Component {
 
     this.ThResultSetStore.initRepository(repoName);
 
-    this.updateUrlFromchange = this.updateUrlFromchange.bind(this);
     this.closeJob = this.closeJob.bind(this);
 
+    this.skipNextPageReload = false;
```

Comment:
I set this as a class member rather than in the ``state`` because it has no bearing on the ``DOM`` rendering, and I don't want to have to kick off a ``state`` lifecycle run prior to a change that will also invoke a ``state`` lifecycle run.  It felt inefficient.  I'll admit I went back and forth on this, but this seemed better.  But open to opinions.  :)

Reply:
Would it work here to add it to the state and use shouldComponentUpdate?

Reply:
That's a good point.  I bet I could make that work, yeah.  Though this line in the docs makes me think perhaps I shouldn't:

    This method only exists as a performance optimization. Do not rely on it to 
    “prevent” a rendering, as this can lead to bugs. Consider using the built-in 
    PureComponent instead of writing shouldComponentUpdate() by hand. 
    PureComponent performs a shallow comparison of props and state, and 
    reduces the chance that you’ll skip a necessary update.

It feels like it might make the rendering a bit brittle if we change which items were in state, then we'd have to consider each of them to determine whether to re-render or not.  Let's consider this for the future if we hit issues with doing it this way.  I may likely be proven wrong in this approach.  :)